### PR TITLE
Chore: add consistent code style

### PR DESCRIPTION
**Pull request by dbmalkovsky**
Add code style files for IntelliJ and Eclipse along with instructions how to import and use.

Once (if) accepted I will start applying the styles to the source files.

### CODE REVIEW

**Comment by WiseEagleOwl** [comment]
I would very much prefer if this was done using a CLI tool with one configuration file as a single source of truth. This would create the opportunity to automate this with a *Github action* later on.
The issue i see here with two configuration files for *Eclipse* and *IntelliJ* is to make sure both IDEs produce the exact same result for all files in this project. 
Additionally we will also need a configuration for *Netbeans* as this is the IDE that is currently used to develop this project. This will make the problem even worse as now three IDEs have to agree on every detail of the source formatting.

**Comment by BlakeTNC** [comment]
Agreed that we should use a tool that will produce identical formatting output in all IDE's.

I was looking around at options. I found some options that are listed here.

### Google Java Style
- formatter: https://github.com/google/google-java-format
- third party integrations: https://github.com/google/google-java-format#third-party-integrations
- This formatter implements the standard java style used by google. (aka "Google Java Style".) The style is predetermined and non-configurable, by design.
- It has maven plugins and github hooks available. I imagine this tool and the style are well supported, since it's google.
- Style notes:
-- It uses 2 space indents instead of 4 spaces. Personally, I -think- I could get used to this. I've always used 4 spaces but after staring at some code, I imagine and hope I could adapt. If I or anyone else here can't adapt, then I suppose we could later undo our style change.
-- After reading a few discussions on "2 spaces vs 4 spaces", I can see acceptable logic behind why they chose 2 spaces.
-- Otherwise, it's pretty similar to our current styling, it think. Importantly (to me) it appears to standardize a 100 column line maximum. I would not have been happy with 80 columns.

### AOSP Android Open Source Project style
- I think the google formatter might be able to be configured to implement this style instead. 
- Uses 4 spaces. 
 
### Arctic style, formatting tool
formatter: http://astyle.sourceforge.net/
I don't know how good tool this is. I'm just noting its existence. 
I think the style rules are configurable with this one, but don't know. 

### A list of common java styles
Is here: https://dev.to/scottshipp/which-java-code-style-should-you-use-1k5a

### My current inclination
If others think it's alright, we could explore or try the google style, with the google formatter. I'm seeking your feedback.

**Comment by dbmalkovsky** [comment]
A series of random observations:

- I looked and found other options too; one was commercial and didn't look very active and the open source ones (other than Google) seemed dormant too.
- It really doesn't matter what style is used: 2 vs. 4 spaces; although I would prefer a longer line length like other projects I work on.
- This project isn't making large new code changes so the impact of a style isn't very big.
- Once the entire project is formatted to a style maintaining it is not very hard; PR code reviews usually in my experience find most style issues.
- I prefer to be able to make and see the "final" format in my IDE before pushing the code; so having IDE plugins is my preference.    NOTE: I don't see a NetBeans plugin.
- Having said that a post CLI action to process the file is fine.
- Using the **AOSP** style noted above to get 4 spaces would be nice.  The CLI and the IntelliJ plugin both support that option.  The CLI uses:
<pre>--aosp, -aosp, -a         
          use AOSP style instead of Google Style (4-space indentation).
</pre>

In summary, any choice/style is better than no consistent code style.

**Comment by BlakeTNC** [comment]
I looked closer at ASOP. It requires variable names to start with m for members, s for static, so that style is a no go. 
Reference: https://stackoverflow.com/questions/26387905/why-not-android-code-style-follow-google-java-code-style

Here is a netbeans plugin  for google style. This plugin appears to be maintained. Because google style is likely a somewhat popular thing, I imagine there are tools for applying that style within eclipse or intellij as well. But I have not looked for such tools for other IDEs. 

It looks like the latest version of "google-java-format" to support running the style plugin under java 8 may have been version 1.7.0.5. That version is implied by one of the answers to this issue: 
https://github.com/google/google-java-format/issues/529#issuecomment-705786156

I would not mind using an older version of google-java-format on my personal ide installations if it works. I imagine an older version could work fine. But that is just a guess.

I should include that I saw this snippet at the following link. "(Note that the plugin can still format Java 8 code. Just the runtime needs to be Java 11+.)" I don't know what exactly that implies for our usage as a github action. 
https://github.com/google/google-java-format/issues/529#issuecomment-705442810

> Having said that a post CLI action to process the file is fine.

Agreed. One simpler solution I can imagine to approximate a "local preview" of the style is:
- Using the built in formatting settings of anyones IDE to "approximate" google style. Ie use two space indents and four space continuations, and possibly any other more detailed settings as desired.
- Then using a separate process to "finalize" the style at some point, either prior to the commit or during the github commit or both. 
- The only advantage of this idea is that it does not require any ide plugins. Just a final github action. 
- Of course anyone who can fine or wants to see the exact syle locally probably has ways to accomplish that too, via ide plugin or the maven plugin, etc. 

Note, all my discussion has just been to help a bit with researching solutions. Someone else with more motivation on the style issue would be implementing this. I imagine as a github process of some sort. I've never worked with github hooks or actions. 

I hope that helps a bit.

**Comment by dbmalkovsky** [comment]
RE: 
I looked closer at ASOP. It requires variable names to start with m for members, s for static, so that style is a no go.
Reference: https://stackoverflow.com/questions/26387905/why-not-android-code-style-follow-google-java-code-style

When I tested the formatter with ASOP style it doesn't enforce those variable names; it just reformats the file.

**Comment by BlakeTNC** [comment]
> When I tested the formatter with ASOP style it doesn't enforce those variable names; it just reformats the file.

That is good to know. I'm thinking I would personally still vote for google style for a few reasons. There are three of us contributors here though. I would be fine with whichever style that 2 out of three people decide they want. (google vs aosp)

my own reasons:
- My guess is that Google style might have wider support across third party ide plugins etc. 
- 2 space indents are growing on me just fine in my coding work the last day or so. 
- 8 space line continuations in AOSP seem extreme to me. 

note: Apparently the acronym is "AOSP" not "ASOP".

**Comment by dbmalkovsky** [comment]
Closing; current topic is being work on in [PR-151](https://github.com/LGoodDatePicker/LGoodDatePicker/pull/151)
